Source/Core/Core/HW/VideoInterface.cpp

```diff
@@ -558,24 +567,12 @@ static void BeginField(FieldType field)
 		// videos in Metroid Prime don't render correctly using this hack.
 		fbStride /= 2;
 		fbHeight *= 2;
-		if (m_VBlankTimingOdd.PRB < m_VBlankTimingEven.PRB)
-		{
-			xfbAddr = GetXFBAddressTop();
-		}
-		else
-		{
-			xfbAddr = GetXFBAddressBottom();
-		}
-	}
-	else
-	{
-		if (field == FieldType::FIELD_EVEN)
-		{
-			xfbAddr = GetXFBAddressTop();
+
+		if ((field == FieldType::FIELD_ODD) && (m_VBlankTimingOdd.PRB == m_VBlankTimingEven.PRB + 1)) {
+			xfbAddr -= (fbStride*2);
```

Comment:
This trades declarative programming style (the code explicitly says what it's doing) against what's undocumented black magic (or maybe it is documented somewhere in the block above, either way it's much less apparent what it's doing).

Is there any chance you can document these lines, or even better - use a declarative code flow here?